motion_computation/src/bsm_to_external_object_convertor.cpp

```diff
@@ -0,0 +1,348 @@
+  tf2::Quaternion R_m_s = R_m_n * R_n_h;  // Rotation of sensor in map frame under assumption that
+                                          // distance from map origin is sufficiently small so as to
+                                          // ignore local changes in NED orientation
+
+  RCLCPP_DEBUG_STREAM(rclcpp::get_logger("motion_computation::conversion"), "R_m_n (x,y,z,w) : ( " << R_m_n.x() << ", " << R_m_n.y() << ", " << R_m_n.z() << ", " << R_m_n.w());
+
+  RCLCPP_DEBUG_STREAM(rclcpp::get_logger("motion_computation::conversion"), "R_n_h (x,y,z,w) : ( " << R_n_h.x() << ", " << R_n_h.y() << ", " << R_n_h.z() << ", " << R_n_h.w());
+
+  // RCLCPP_DEBUG_STREAM(rclcpp::get_logger("motion_computation::conversion"), "R_h_s (x,y,z,w) : ( " << R_h_s.x() << ", " << R_h_s.y() << ", " << R_h_s.z() << ", " << R_h_s.w());
+
+  RCLCPP_DEBUG_STREAM(rclcpp::get_logger("motion_computation::conversion"), "R_m_s (x,y,z,w) : ( " << R_m_s.x() << ", " << R_m_s.y() << ", " << R_m_s.z() << ", " << R_m_s.w());
+
+  tf2::Transform T_m_s(R_m_s, tf2::Vector3(map_point.x(), map_point.y(),
+                                           map_point.z()));  // Reported position and orientation
+                                                             // of sensor frame in map frame
+
+        tf2::Transform T_m_n_no_heading(R_m_n, {0, 0, 0}); // Used to transform the covariance
+
+        // This covariance represents the covariance of the NED frame N-lat,
+        // E-lon, heading- angle east of north This means that the covariance is
+        // in the NED frame, and needs to be transformed to the map frame
+        // clang-format off
+        std::array<double, 36> input_covariance = { 
+          lat_variance, 0, 0,  0, 0, 0,
+          0, lon_variance, 0,  0, 0, 0,
+          0, 0, 0,  0, 0, 0,
+          0,  0,  0,  1,  0, 0,
+          0,  0,  0,  0,  1, 0, 
+          0,  0,  0,  0,  0, heading_variance
+        };
+        // clang-format on
+
+        std::array<double, 36> new_cov = tf2::transformCovariance(input_covariance,
+                                     // Per the usage of transformCovariance in tf2_geometry_msgs
+                                     // this frame should be the transform between the map which the pose
+                                     // is in an the target frame.
+                                     T_m_n_no_heading);
+
+        // Populate message
+        pose.pose.position.x = T_m_s.getOrigin().getX();
+        pose.pose.position.y = T_m_s.getOrigin().getY();
+        pose.pose.position.z = T_m_s.getOrigin().getZ();
+
+        pose.pose.orientation.x = T_m_s.getRotation().getX();
+        pose.pose.orientation.y = T_m_s.getRotation().getY();
+        pose.pose.orientation.z = T_m_s.getRotation().getZ();
+        pose.pose.orientation.w = T_m_s.getRotation().getW();
+
+        pose.covariance = new_cov;
+
+        return pose;
+}
+
+std::vector<carma_perception_msgs::msg::PredictedState> predicted_poses_to_predicted_state(
```

Comment:
Rather than duplicating this function, can you just include the psm helper and call it directly